kiwi/bootloader/config/grub2.py

```diff
@@ -187,8 +187,10 @@ def setup_sysconfig_bootloader(self):
         * FAILSAFE_APPEND
         """
         sysconfig_bootloader_entries = {
-            'LOADER_TYPE': self.boot_directory_name,
-            'LOADER_LOCATION': 'mbr'
+            'LOADER_TYPE':
+                'grub2-efi' if self.firmware.efi_mode() else 'grub2',
+            'LOADER_LOCATION':
+                'mbr'
```

Comment:
This looks wrong, on EFI there is no MBR.

I'm not sure where LOADER_LOCATION is used though, there doesn't seem to be sufficient documentation.

Reply:
I agree it's wrong for efi and it also could be wrong for bios if e.g installed into a partition. This value exists only because in the past perl-bootloader and iirc yast required a value. I would be more than happy to delete it entirely but wasn't sure if it cause any unwanted after effects. The reason why you see it in the diff is just because the indentation was changed by me. So it's present since a long time.

I would be ok deleting it and take the risk to see if something breaks. In this case we finally have a chance to know why it exists, add some comments and do the right thing. So far nobody really knows.

Thoughts ?

Reply:
Sounds like a good idea - but would fit better into a separate PR which is not part of an important last-minute bugfix. We shouldn't risk breaking that submission.

Reply:
agreed, I'll open an issue so we don't forget